packages/nimble-components/src/table/tests/table.spec.ts

```diff
@@ -864,6 +870,465 @@ describe('Table', () => {
                 verifyRenderedData(sortedData);
             });
         });
+
+        describe('hierarchical data', () => {
+            const hierarchicalData: SimpleTableRecord[] = [
+                {
+                    stringData: 'Parent 1',
+                    stringData2: 'Parent 1',
+                    numericData: 0,
+                    moreStringData: 'foo',
+                    id: '0'
+                },
+                {
+                    stringData: 'Parent 1 Child',
+                    stringData2: 'Parent 2 Child',
+                    numericData: 0,
+                    moreStringData: 'foo',
+                    id: '1',
+                    parentId: '0',
+                    parentId2: 'Parent 2',
+                    parentId3: 'Parent 1'
+                },
+                {
+                    stringData: 'Parent 1 Grandchild',
+                    stringData2: 'Parent 1 Grandchild',
+                    numericData: 0,
+                    moreStringData: 'foo',
+                    id: '2',
+                    parentId: '1',
+                    parentId2: 'Parent 1 Child',
+                    parentId3: 'Parent 2 Child'
+                },
+                {
+                    stringData: 'Top Level No Child',
+                    stringData2: 'Top Level No Child',
+                    numericData: 0,
+                    moreStringData: 'foo',
+                    id: '3'
+                },
+                {
+                    stringData: 'Parent 2',
+                    stringData2: 'Parent 2',
+                    numericData: 0,
+                    moreStringData: 'foo',
+                    id: '4'
+                },
+                {
+                    stringData: 'Parent 2 Child',
+                    stringData2: 'Parent 1 Child',
+                    numericData: 0,
+                    moreStringData: 'foo',
+                    id: '5',
+                    parentId: '4',
+                    parentId2: 'Parent 1',
+                    parentId3: 'Parent 2'
+                },
+                {
+                    stringData: 'Parent 1 Child 2',
+                    stringData2: 'Parent 1 Child 2',
+                    numericData: 0,
+                    moreStringData: 'foo',
+                    id: '6',
+                    parentId: '0',
+                    parentId2: 'Parent 1',
+                    parentId3: 'Parent 2'
+                }
+            ];
+            it('shows collapse all button with hierarchical data', async () => {
```

Comment:
Could you make sure we have tests that verify the behavior of clicking the collapse all button with hierarchical data?